committer/src/setup.rs

```diff
@@ -74,6 +79,191 @@ pub fn block_committer(
+    let block_bundler = block_bundler(
+        fuel.clone(),
+        storage.clone(),
+        cancel_token.clone(),
+        config,
+        registry,
+    );
+
+    let fee_api = CachingApi::new(
+        ethereum_rpc.clone(),
+        internal_config.l1_blocks_cached_for_fee_metrics_tracker,
+    );
+
+    let fee_metrics_updater_handle =
+        ethereum_da_fee_metrics_tracker(fee_api.clone(), cancel_token.clone(), config, registry)?;
+
+    let committer = state_committer(
+        fuel.clone(),
+        ethereum_rpc.clone(),
+        storage.clone(),
+        cancel_token.clone(),
+        config,
+        registry,
+        fee_api,
+    )?;
+
+    let listener = state_listener(
+        ethereum_rpc,
+        storage.clone(),
+        cancel_token.clone(),
+        registry,
+        config,
+        last_finalization_metric(),
+    );
+
+    let state_importer_handle = block_importer(
+        fuel,
+        storage.clone(),
+        cancel_token.clone(),
+        config,
+        internal_config,
+        registry,
+    );
+
+    let state_pruner_handle = state_pruner(storage.clone(), cancel_token.clone(), registry, config);
+
+    let handles = vec![
+        committer,
+        state_importer_handle,
+        block_bundler,
+        listener,
+        fee_metrics_updater_handle,
+        state_pruner_handle,
+    ];
+
+    Ok(handles)
+}
+
+pub async fn eigen_da_services(
+    fuel: FuelApi,
+    storage: Database,
+    cancel_token: CancellationToken,
+    config: &config::Config,
+    internal_config: &config::Internal,
+    registry: &Registry,
+) -> Result<Vec<tokio::task::JoinHandle<()>>> {
+    let block_bundler = eigen_block_bundler(
+        fuel.clone(),
+        storage.clone(),
+        cancel_token.clone(),
+        config,
+        registry,
+    );
+
+    let (state_committer_handle, state_listener_handle, fee_metrics_handle) =
+        match config.da_layer.clone() {
+            Some(DALayer::EigenDA(eigen_config)) => {
+                let eigen_da = eigen_adapter(&eigen_config, internal_config).await?;
+                let committer = eigen_state_committer(
+                    fuel.clone(),
+                    eigen_da.clone(),
+                    storage.clone(),
+                    cancel_token.clone(),
+                    config,
+                    registry,
+                )?;
+
+                let listener = eigen_state_listener(
+                    eigen_da.clone(),
+                    storage.clone(),
+                    cancel_token.clone(),
+                    config,
+                    registry,
+                    last_finalization_metric(),
+                )?;
+
+                let fee_metrics_handle =
+                    eigen_fee_metrics_tracker(eigen_da, cancel_token.clone(), config, registry)?;
+
+                (committer, listener, fee_metrics_handle)
+            }
+            _ => {
+                return Err(Error::Other("Invalid da layer config".to_string()));
```

Comment:
nit: error message should provide more context about what configuration was expected versus what was received

```suggestion
                return Err(Error::Other(format!(
                    "Expected EigenDA configuration but received: {:?}", config.da_layer
                )));
```